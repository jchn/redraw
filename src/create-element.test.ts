import createElement from './create-element'

describe('createElement', () => {
  it('should return a vnode', () => {
    const vnode = createElement('rectangle', { foo: 'bar' })

    expect(vnode.type).toEqual('rectangle')
    expect(vnode.props).toEqual({ foo: 'bar', children: [] })
    expect(vnode).toHaveProperty('_depth', 0)
    expect(vnode).toMatchSnapshot()
  })
})
