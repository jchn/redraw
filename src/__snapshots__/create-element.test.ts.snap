// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createElement should return a vnode 1`] = `
Object {
  "_children": null,
  "_component": null,
  "_depth": 0,
  "_dimensions": null,
  "_matrix": Float32Array [
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  "_parent": null,
  "_position": null,
  "props": Object {
    "children": Array [],
    "foo": "bar",
  },
  "type": "rectangle",
}
`;
